Board.java

```diff
@@ -30,33 +28,123 @@ public Board playerMakesMove(char player, int position) {
         return this;
     }
 
+    public String getWinner() {
+        for (String combination : allWinningCombinations()) {
+            if (!combination.contains("x") && !combination.contains("-")) {
+                return "o";
+            }
+            if (!combination.contains("o") && !combination.contains("-")) {
+                return "x";
+            }
+        }
+        return null;
+    }
```

Comment:
Avoid returning null or using null as much as possible, and definitely don't have a condition based on it as in hasWinner() below.  Opt for getWinner == 'o' || getWinner() == 'x' over getWinner() != null